rfc/0067-extended-sdk-observability-rto-v2.md

```diff
@@ -596,90 +605,104 @@ This attribute is a standard OpenTelemetry attribute and should be placed on N1Q
 | Views | Not present |
 | Analytics | The actual analytics query |
 
+Sanitization should be performed on the statement to exclude sensitive information, by replacing all literals (e.g. String, Numeric, Boolean, Date and Time) with a placeholder, `?`.
```

Comment:
If we're going to do sanitization can you please include a regex or some pseudocode of how that should work? Otherwise we're going to end up with SDKs doing different things, and having different bugs.

Reply:
Yes, this feels a bad path.  The SDKs do not parse the query string, it's too complex.

We could only include the statement if the user provided 1 or more named or positional parameters?  That captures the spirit of what Otel says:

> Parameterized query text SHOULD NOT be sanitized. Even though parameterized query text can potentially have sensitive data, by using a parameterized query the user is giving a strong signal that any sensitive data will be passed as parameter values, and the benefit to observability of capturing the static part of the query text by default outweighs the risk.

Reply:
Updated to say that the statement should only be included if named/positional parameters are used instead.